src/js/components/sharedComponents/header/DropdownItem.jsx

```diff
@@ -6,53 +6,85 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
+import * as redirectHelper from 'helpers/redirectHelper';
+
 import DropdownComingSoon from './DropdownComingSoon';
 
 const propTypes = {
     url: PropTypes.string,
     label: PropTypes.string,
     enabled: PropTypes.bool,
     newTab: PropTypes.bool,
-    isFirst: PropTypes.bool
+    isFirst: PropTypes.bool,
+    externalLink: PropTypes.bool
 };
 
-const DropdownItem = (props) => {
-    let className = 'nav-children__link_disabled';
-    let comingSoon = (
-        <div className="nav-children__coming-soon">
-            <DropdownComingSoon />
-        </div>
-    );
-    if (props.enabled) {
-        className = '';
-        comingSoon = null;
-    }
+export default class DropdownItem extends React.Component {
+    constructor(props) {
+        super(props);
 
-    const newTabProps = {};
-    if (props.newTab) {
-        newTabProps.target = '_blank';
-        newTabProps.rel = 'noopener noreferrer';
+        this.redirect = this.redirect.bind(this);
     }
 
-    let firstClass = '';
-    if (props.isFirst) {
-        firstClass = 'nav-children__list-separator_hidden';
+    redirect() {
+        redirectHelper.showRedirectModal(this.props.url);
     }
 
-    return (
-        <li className="nav-children__list-item">
-            <hr className={`nav-children__list-separator ${firstClass}`} />
+    render() {
+        let className = 'nav-children__link_disabled';
+        let comingSoon = (
+            <div className="nav-children__coming-soon">
+                <DropdownComingSoon />
+            </div>
+        );
+
+        if (this.props.enabled) {
+            className = '';
+            comingSoon = null;
+        }
+
+        let link = (
             <a
                 className={`nav-children__link ${className}`}
-                href={props.url}
-                {...newTabProps}>
-                {props.label}
+                href={this.props.url}>
+                {this.props.label}
                 {comingSoon}
             </a>
-        </li>
-    );
-};
+        );
 
-DropdownItem.propTypes = propTypes;
+        if (this.props.enabled && this.props.newTab && !this.props.externalLink) {
+            link = (
+                <a
+                    className="nav-children__link"
+                    href={this.props.url}
+                    target="_blank"
+                    rel="noopener noreferrer">
+                    {this.props.label}
+                </a>
+            );
+        }
+        else if (this.props.enabled && this.props.externalLink) {
```

Comment:
What is the use-case this is handling?

Reply:
The Agency Submission Files link opens in a new tab but does not trigger the modal. This handles the Database Snapshots link, which does trigger the modal.